src/dataframerow/utils.jl

```diff
@@ -212,11 +210,12 @@ function row_group_slots(cols::NTuple{N,<:Union{CategoricalVector,PooledVector}}
         end
         vals = map((m, r, s) -> m[r+1] * s, refmaps, refs, strides)
         j = sum(vals) + 1
-        if skipmissing
-            j = any(x -> x < 0, vals) ? 1 : j + 1
+        if skipmissing && any(x -> x < 0, vals)
```

Comment:
Can you please comment when `vals` can take a negative value?

Reply:
Yeah, that indicates a missing value in the corresponding column. I'll add a comment.

Reply:
This is what I assumed, but my question is how does this happen (i.e. what is the source of this negative value). For this to happen either `m[r+1]` or `s` should be negative. I understand that `s` is always non-negative, and `m` is `refmaps` where I thought we are non-negative. For sure I miss something, but I am asking so that in the future I understand this part of code better.

Reply:
Well the code above does all kinds of weird computations so that `refmaps` has a `-1` entry for `missing` (the first entry for `CategoricalArray`, an arbitrary entry for `PooledArray`). Do you think a comment is needed above?

Reply:
Yes, I think ideally a comment how the contents of `remaps` should be interpreted would be ideal. Thank you.